myapp/src/stores/TodoListStore.js

```diff
@@ -0,0 +1,63 @@
+import { observable, action, computed, configure } from "mobx";
+import todoListItem from "./../todoListItem.json";
+
+configure({ enforceActions: true })
+
+
+class TodoListStore {
+    @observable listItems = todoListItem;
+
+    @action
+    handleChangeCheckBox = (id) => {
+        const newTodoItems = this.listItems.map((element) => {
+            if (id === element.id) {
+                return {
+                    ...element,
+                    done: !element.done
+                }
+            }
+            return element;
+        });
+        this.listItems = newTodoItems;
+
+    }
+    //    handleChangeCheckBox = (index) => {
+    //     const item = [...this.listItems]
+    //     item[index].done = !item[index].done
+    //     this.listItems[index].done = !this.listItems[index].done;
+    // }
+
+    @action
+    onDelete = (index) => {
+        const todosList = [...this.listItems]
+        todosList.splice(index, 1)
+        this.listItems = todosList;
+    }
+
+    @action
+    onAdd = (event) => {
+        event.preventDefault();
+        const newTodo = {
+            "id": this.listItems.length + 1,
```

Comment:
the Idea of `id` is to set a unique property for the object!
So the length in sommepoint won't be unique think about this:
>you have 3 items the length is 3 ids: 1, 2, 3
>you deleted a one in the middle so your ids now: 1, 3
>you added a new one your ids: 1, 3, 3

see? 🙃